organizations/data.py

```diff
@@ -313,76 +334,113 @@ def bulk_create_organization_courses(organization_course_pairs):
             Organization-course linkages that DO already exist will be activated, if inactive.
 
             Assumption: All provided organizations already exist in the DB.
-    """
-    # For sake of having sane variable names, please understand
-    #   * "orgslug" to mean "lowercase organization short name" and
-    #   * "courseid" to mean "stringified course run key"
-    # in the context of this function.
-    # We normalize short_names to lowercase because MySQL UNIQUE is case-insensitive.
 
-    # Build set of pairs: (lowercased org short name, course key string).
-    orgslug_courseid_pairs = {
+        dry_run (bool):
+            Optional, defaulting to False.
+            If True, log organizations-course linkaages that would be created or
+            activated, but do not actually apply the changes to the database.
+    """
+    def linkage_to_pair(linkage):
+        """
+        Given a `OrganizationCourse` (aka org-course linkage) instance,
+        return a pair of strings:
+        (lowercased organization short name, stringified course key)
+
+        We normalize short_names to lowercase because MySQL UNIQUE is case-insensitive.
+        """
+        return linkage.organization.short_name.lower(), linkage.course_id
+
+    # Grab all org-course linkages from db (whether active or inactive).
+    db_linkages = list(
+        internal.OrganizationCourse.objects.all().prefetch_related(
+            'organization'
+        )
+    )
```

Comment:
So this is a heavier on the DB side than the old code was (2 queries with all columns, instead of 1 query with fewer columns), but I think it's well worth the readability boost.

FWIW, I expect the `organizations_organizationcourse` table to have on the order of 10k rows, and `organizations_organization` table to have on the order of 500 rows.